lib/src/camera.dart

```diff
@@ -73,7 +78,24 @@ class CameraManager {
       collection.videoServer.sendMessage(VideoData(details: details));
       timer?.cancel();
     } else {
-      collection.videoServer.sendMessage(VideoData(frame: frame.data, details: CameraDetails(name: CameraName.ROVER_FRONT, status: CameraStatus.CAMERA_ENABLED)));
+      
+      if(frame.data.length < 60000){
+        collection.videoServer.sendMessage(VideoData(frame: frame.data, details: details));
+      } else {
+        details.status = CameraStatus.FRAME_TOO_LARGE;
+        collection.videoServer.sendMessage(VideoData(details: details));
+        details.status = CameraStatus.CAMERA_ENABLED;
+        if(details.quality > 80){
+          details.quality--;
+        } else if(details.resolutionHeight > 500){
+          details.resolutionHeight--;
+        } else if(details.resolutionWidth > 500){
+          details.resolutionWidth--;
+        } else {
+          // THROW SOME ERROR?
+        }
```

Comment:
- the quality decrease is a much faster way to get a small image. Like you can get an HD image with like 5% quality and it should work, so let's just stick with that instead of changing resolution
- let's keep an eye on how this works when testing -- if the quality gets drastically reduced, the user should know why. -1% * ~20 FPS ≈ 1.5 seconds until the issue is resolved which should be enough time for the dashboard to get the FRAME_TOO_LARGE
- I've seen errors above 75% -- and higher resolutions would need an even lower quality -- so how about you lower the threshold down to like 50. Or even 0 (still visible just no color). This should only happen anyway if the current image just can't be sent so it's not the end of the world to ruin the frame. At least you save the camera
- As for the last else, just set the status to FRAME_TOO_LARGE without changing it back. That way the next iteration will stop reading the camera and the dashboard will keep the warning up